src/psyclone/dynamo0p3.py

```diff
@@ -6156,6 +6157,25 @@ def __init__(self):
         self._cma_operation = None
         self._is_intergrid = False  # Whether this is an inter-grid kernel
 
+    def reference_accesses(self, var_accesses):
+        '''Get all variable access information. All accesses are marked
+        according to the kernel declaration.
+        :param var_accesses: \
+            :py:class:`psyclone.core.access_info.VariablesAccessInfo`
+        '''
+        for arg in self.arguments.args:
+            # If arg is a grid type argument, it does not have
+            # the access mode defined. So take it from the arg_descriptor:
+            if arg.vector_size > 0:
+                # It's an array, so add an arbitrary index value for the
+                # stored indices (which is at this stage the only way to
+                # indicate an array access).
+                var_accesses.add_access(arg.name, arg.access, self, [1])
+            else:
+                var_accesses.add_access(arg.name, arg.access, self)
+        super(DynKern, self).reference_accesses(var_accesses)
+        var_accesses.next_location()
```

Comment:
Please could you explain what this line is for (in a comment).

Reply:
Done